checker.c

```diff
@@ -1,21 +1,17 @@
 #include <stdio.h>
 #include <assert.h>
+#include "header.h"
+
+int count;
 
 int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+  count = isTempNotInRange(temperature);
+  count = isStateNotInRange(soc);
+  count = isChargeNotInRange(chargeRate);
+  return count;
 }
 
 int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
+  assert(batteryIsOk(25, 70, 0.7)==0);
+  assert(batteryIsOk(50, 85, 0)>0);
```

Comment:
Any calls to `batteryIsOk` after this point will give >0 (even if you pass everything within range)
To avoid such issues, avoid use of globals. To accumulate the checking results, make a structure and pass a pointer to it.